quick_start/kusto_sample_app.py

```diff
@@ -0,0 +1,261 @@
+
+
+def main():
+    print("kusto sample app is starting")
+
+    # Todo - Learn More: For additional information on how to authorize users and apps on Kusto Database see:
+    #  https://docs.microsoft.com/en-us/azure/data-explorer/manage-database-permissions
+    if authenticationMode == "deviceCode":
+        print("During the run you will be prompted by the browser to enter a device code.")
+        print("Return to the console of the sample app and copy the code to provide from it.")
+        input("Press enter to continue...")
+
+    # Todo - Tip: Avoid creating a new Kusto Client for each use.
+    #  Create the clients once and use them as long as possible.
+    kusto_connection_string = create_connection_string(kustoUri, authenticationMode)
+    ingest_connection_string = create_connection_string(ingestUri, authenticationMode)
+    kusto_client = KustoClient(kusto_connection_string)
+    ingest_client = QueuedIngestClient(ingest_connection_string)
+
+    # Todo - Learn More: For additional information on how to create tables see: https://docs.microsoft.com/en-us/azure/data-explorer/one-click-table
+    print("")
+    print(f"Creating table '{databaseName}.{tableName}' if needed:")
+    # Todo (Yochai) what's the table creation command
+    command = ".show version"
+    if not run_control_command(kusto_client, databaseName, command):
+        print("Failed to create or validate table exists.")
+        exit(-1)
+
+    # Todo - Learn More: For additional information on Kusto Query Language see: https://docs.microsoft.com/en-us/azure/data-explorer/write-queries
+    print("")
+    print(f"Initial row count for '{databaseName}.{tableName}' is:")
+    run_query(kusto_client, databaseName, f"{tableName} | summarize count()")
+
+    # Todo - Learn More:
+    #  Kusto batches data for ingestion efficiency. The default batching policy ingests data when one of the following conditions are met:
+    #   1) more then a 1000 files were queued for ingestion for the same table by the same user
+    #   2) more then 1GB of data was queued for ingestion for the same table by the same user
+    #   3) More then 5 minutes have passed since the first file was queued for ingestion for the same table by the same user
+    #  In order to speed up this sample app, we attempt to modify the default ingestion policy to ingest data after 10 seconds have passed
+    #  For additional information on customizing the ingestion batching policy see:
+    #   https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/batchingpolicy
+    #  You may also skip the batching for some files using the FlushImmediatly property, though this option should be used with care as it is inefficient
+    print("")
+    print(f"Altering the batching policy for '{tableName}'")
+    batching_policy = '{ "MaximumBatchingTimeSpan": "00:00:10", "MaximumNumberOfItems": 500, "MaximumRawDataSizeMB": 1024 }'

```

Comment:
might as well have maximum items = 1

Reply:
I would rather not steer them in a direction of a policy which I would not recommend.
There's also some info on ingestion batching to help them understand which direction they should actually take.